### PR TITLE
Add build info metric

### LINKED ISSUE #1132: Add Support for kube_state_metrics_build_info
This is a FEATURE REQUEST.

Please add support for kube_state_metrics_build_info as part of the metrics exposed on telemetry port 8081. 
kube_state_metrics_build_info shall include the version of the kube-state-metrics.
Most Prometheus exporters are currently supporting such metric.

### PR DESCRIPTION
**Pull request by yeya24**
Signed-off-by: Ben Ye <yb532204897@gmail.com>

**What this PR does / why we need it**:

This pr adds the `kube_state_metrics_build_info` metric as required by #1132 

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1132

### CODE REVIEW

File: pkg/version/version.go [hunk 1]
```diff
@@ -18,6 +18,9 @@ import (
 	"os"
 	"path/filepath"
 	"runtime"
+
+	"github.com/prometheus/client_golang/prometheus"
```
Comment (tariq1890) [inline pkg/version/version.go:22]: let's not add these imports to the `version` pkg

File: pkg/version/version.go [hunk 2]
```diff
@@ -62,3 +65,18 @@ func GetVersion() Version {
 		Platform:  fmt.Sprintf("%s/%s", runtime.GOOS, runtime.GOARCH),
 	}
 }
+
+// RegisterBuildInfo registers the build info metric.
+func RegisterBuildInfo(r prometheus.Registerer) {
```
Comment (tariq1890) [inline pkg/version/version.go:70]: We can move this method to main.go instead. In `main.go`, we can call `version.GetVersion()`, get the version struct and use its fields to instantiate the Gauge vector in  `main.go`

**Comment by tariq1890** [comment]
Thanks so much for this PR @yeya24 . Provided my comments

File: pkg/version/version.go (hunk 2 above)
Comment (yeya24) [reply pkg/version/version.go:70]: Yes, thanks! I will update it.

**Comment by lilic** [comment]
Thanks for the PR, I already started the kube-state-metrics alpha 2.0 release, so might need to wait a bit longer until that is cut. Hope that's okay. :)

**Comment by yeya24** [comment]
I just updated the pr, please take a look again @tariq1890 

Hi @lilic, no problem. Thanks for letting me know :+1:

**Comment by YuikoTakada** [comment]
Thank you for the PR. This new metrics need to be added to docs also? I'm not sure.

**Comment by yeya24** [comment]
> Thank you for the PR. This new metrics need to be added to docs also? I'm not sure.

Yes, you are totally right. Thank you! Let me update the pr.

**Comment by lilic** [comment]
@yeya24 I changed the base to release-2.0, do you mind also adding the docs to the README like we have for other ksm own metrics and we can merge afterwards, thanks! :)

File: main.go [hunk 3]
```diff
@@ -295,3 +297,18 @@ func buildMetricsServer(m *metricshandler.MetricsHandler, durationObserver prome
 	})
 	return mux
 }
+
+func registerBuildInfo(r prometheus.Registerer) {
+	v := version.GetVersion()
+	promauto.With(r).NewGaugeFunc(prometheus.GaugeOpts{
+		Name: "kube_state_metrics_build_info",
```
Comment (tariq1890) [inline main.go:304]: Do we need it to be `"kube_state_metrics_build_info"`, what about just `build_info`? We have the job label to specify that it pertains to kube-state-metrics right?

**Comment by yeya24** [comment]
Sorry for the late, I just updated the doc. Please take a look.

File: main.go (hunk 3 above)
Comment (yeya24) [reply main.go:304]: I prefer the full name `kube_state_metrics_build_info`

**Comment by lilic** [comment]
Currently right in the middle of cutting the next release, but we can merge straight after that. Bad timing, sorry :) 
/hold

**Comment by yeya24** [comment]
Hi, how can I push forward this pr? Do I need to change the branch to `master`?

**Comment by lilic** [comment]
Yes master sounds good, sorry for the delay! :)

**Comment by yeya24** [comment]
Will close a new pr against master. Sorry for the late.

**Pull request closed without merging**